src/Illuminate/Encryption/Encrypter.php

```diff
@@ -289,4 +311,35 @@ public function getKey()
     {
         return $this->key;
     }
+
+    /**
+     * Get the current encryption key and all previous encryption keys.
+     *
+     * @return array
+     */
+    public function getAllKeys()
+    {
+        return [$this->key, ...$this->previousKeys];
+    }
+
+    /**
+     * Set the previous / legacy encryption keys that should be utilized if decryption fails.
+     *
+     * @param  array  $key
+     * @return $this
+     */
+    public function previousKeys(array $keys)
+    {
+        foreach ($keys as $key) {
+            if (! static::supported($key, $this->cipher)) {
+                $ciphers = implode(', ', array_keys(self::$supportedCiphers));
+
+                throw new RuntimeException("Unsupported cipher or incorrect key length. Supported ciphers are: {$ciphers}.");
```

Comment:
This exception will be ambiguous, as it doesn't indicate _which_ key is invalid. 

It can't return the actual key though - that'd be horribly insecure - however it could return the index in the previous keys array, which would point in the right direction.

Something like this might work:

```
foreach ($keys as $index => $key) {
    if (! static::supported($key, $this->cipher)) {
        $ciphers = implode(', ', array_keys(self::$supportedCiphers));

        throw new RuntimeException("Unsupported cipher or incorrect key length of previous key {$index}. Supported ciphers are: {$ciphers}.");
```